### PR TITLE
Ongoing/settledbattles

### PR DESCRIPTION
**Pull request by eduardavercosa**
## Objective and Motivation
- User can see the battles that have ended (settled) and the battles still opened (ongoing)

### CODE REVIEW

**Comment by rsarai** [comment]
Two things, you need to change the base branch to be `stable`, or we will merge this on a closed branch. Also, there's a conflict on `backend/battling/views.py`

File: backend/battling/templatetags/battling_tags.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from django import template
+
+from ..models import Team
```
Comment (rsarai) [inline backend/battling/templatetags/battling_tags.py]: Let's change this to be an absolute import?

File: backend/battling/views.py [hunk 2]
```diff
@@ -54,61 +49,56 @@ def form_valid(self, form):
...
+
+class DetailBattle(DetailView):
+    template_name = "battling/battle_detail.html"
+    model = Battle
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         battle = self.get_object()
 
         creator = Team.objects.filter(battle=battle, trainer=battle.creator.id)
```
Comment (rsarai) [inline backend/battling/views.py]: This can be a `get`

File: backend/battling/views.py [hunk 3]
```diff
@@ -54,61 +49,56 @@ def form_valid(self, form):
...
-        creator_pokemon = PokemonTeam.objects.filter(team=creator[0])
-
-        context["creator_team"] = [
-            creator_pokemon[0].pokemon,
-            creator_pokemon[1].pokemon,
-            creator_pokemon[2].pokemon,
-        ]
+        context["creator_team"] = creator[0].pokemons.all()
 
         opponent = Team.objects.filter(battle=battle, trainer=battle.opponent.id)
```
Comment (rsarai) [inline backend/battling/views.py]: This can be a `get`

File: backend/templates/battling/battle_detail.html [hunk 4]
```diff
@@ -1,12 +1,16 @@
...
+{% load battling_tags %}
 
 {% block content %}
 <div class="teste">
   <h1>Result</h1>
-  <h3>{{battle.winner.email}} won!</h3>
 
-  {{battle.creator.email}} team:
+  {% if battle.winner %}
+  <h3>{{battle.winner}} won!</h3>
```
Comment (rsarai) [inline backend/templates/battling/battle_detail.html]: Let's keep the same pattern of indentation on the templates

File: backend/templates/battling/ongoing_battles.html [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+      margin: 0;
+      font-family: rubik;
+    }
+  </style>
+</head>
+
+<body>
+  <div>
+    <h1>Ongoing Battles</h1>
+    {% if object_list.all %}
```
Comment (rsarai) [inline backend/templates/battling/ongoing_battles.html:24]: Let's keep the same pattern of indentation on the templates

File: backend/battling/views.py [hunk 6]
```diff
@@ -54,61 +49,56 @@ def form_valid(self, form):
...
+        )
+        context["invited_battles"] = Battle.objects.filter(status="ONGOING").filter(
+            opponent=self.request.user
+        )
+        return context
+
+
+class DetailBattle(DetailView):
+    template_name = "battling/battle_detail.html"
+    model = Battle
```
Comment (rsarai) [inline backend/battling/views.py]: These two views have similar templates, do we need two separate views for this?

File: backend/templates/battling/settled_battles.html [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+      font-family: rubik;
+    }
+  </style>
+</head>
+
+<body>
+  <div>
+    <h1>Settled Battles</h1>
+    {% if object_list.all %}
+    {% for instance in object_list %}
```
Comment (rsarai) [inline backend/templates/battling/settled_battles.html]: Let's keep the same pattern of indentation on the templates

**Review by rsarai — CHANGES_REQUESTED** [review]

File: backend/battling/views.py (hunk 6 above)
Comment (eduardavercosa) [reply backend/battling/views.py]: The DetailBattle is to show each pokemon and its attributes, the ongoing battles just show a list of the battles. The context is just so I can separate the battles created from the invited

File: backend/templates/battling/settled_battles.html (hunk 7 above)
Comment (eduardavercosa) [reply backend/templates/battling/settled_battles.html]: I try, but when I do indentation, vs code itself changes it however it wants to :(

Comment (rsarai) [reply backend/templates/battling/settled_battles.html]: The ideal is to disable whatever it is updating the templates

**Review by rsarai — APPROVED** [review]

**Pull request merged**